### ISSUE TITLE
persistent-mysql deprecation warnings

### ISSUE THREAD

**Issue by darthdeus** [yesodweb/persistent#355]
While trying to figure out how to use `persistent-mysql` I found out that `Connection` is deprecated in favor of `SqlBackend`, and is actually defined as a type alias, though it is still used in type declarations in persistent-mysql (https://github.com/yesodweb/persistent/blob/master/persistent-mysql/Database/Persist/MySQL.hs#L87).

I'm not sure if I'm missing something, or if this can be just changed to `SqlBackend`?

**Comment by snoyberg** [comment]
Looks like it could just be changed.

**Comment by darthdeus** [comment]
Thanks! I have a related question though. By using `withMySQLConn defaultConnectInfo`, I can get `(SqlBackend -> SqlPersistM a) -> SqlPersistM a`. If I have an expression to run such as `runMigration migrateAll :: SqlPersistM ()`, I'd need a way to connect the two pieces together.

At first I thought I could use `runSqlConn` or `runSqlPersistM`, but these are only `SqlPersistM a -> SqlBackend -> IO a`, which isn't compatible with the `withMySQLConn`, which takes a function.

Following [the book](http://www.yesodweb.com/book/persistent), it seems something like this should be possible

```haskell
withMySQLConn defaultConnectInfo $ \c -> liftIO $ runSqlConn (runMigration migrateAll) c
````

but at this point I just get a `SqlPersistM ()`, which I can't run with `runSqlPersistM`, since that requires `SqlBackend` again, and there is no way to get to the backend, except with `withMySQLConn` ... which is what caused the problem in the beginning.

Is there some way in which I'm misusing the `withMySQLConn`? It seems that this should be rather easy, but I can't find a way for the types to align.

**Comment by snoyberg** [comment]
I think you're overly constraining `withMySQLConn`; it will work with *any*
instance of `MonadBaseControl`. The kind of code you've shown above should
work. I can't be certain why it's not without seeing the error messages
you're getting. But I don't think this issue is the right place to discuss
such things.

**Comment by darthdeus** [comment]
Thanks for replying, I've written up everything in [this StackOverflow question](https://stackoverflow.com/questions/28270754/how-can-i-query-using-persistent-mysql-using-withmysqlconn). I'd be really happy if you could point me in the right direction.

**Comment by gregwebs** [comment]
@darthdeus is there something that should be done for this issue or should it be closed?

**Comment by darthdeus** [comment]
@gregwebs looks like it was fixed in #371, closing the issue
